SpriteWander/entity/Entity.cs

```diff
@@ -165,27 +133,36 @@ public void Tick(double time)
             {
                 case AnimEvent.End:
                     NextAnim();
-                    turn(0.7);
+                    TurnRandom(0.7);
                     break;
                 case AnimEvent.Reset:
-                    turn(0.1);
+                    TurnRandom(0.1);
                     if (cycle == 0) NextAnim();
                     else
                     {
                         Event = AnimEvent.Nothing;
                         animTimer = 0;
                         cycle--;
                     }
+                    if (state == Animation.Walk) // Calculate direction to position before moving
+                    {
+                        double angle = Math.Atan2(targetY - Y, targetX - X) * (4.0 / Math.PI) + 2.25;
+                        subState = (Direction)((Math.Round(angle + 7) % 7));
```

Comment:
`Math.Round(angle + 7) % 7` -> `Math.Round(angle + 8) % 8`

The mod opérator give a number between -n and n excluded, to have a number between 0 and 7 included, we need to do +8 % 8, right now the pokemons can't walk diagonolly down left.

Reply:
i will do it in the main branch

Reply:
oopsie daisies